audio/setup.py

```diff
@@ -53,6 +54,67 @@ def find_meta(meta):
     raise RuntimeError("Unable to find __{meta}__ string.".format(meta=meta))
 
 
+def get_recent_changelog(filename):
+    """Grab the entry for the latest release."""
+    contents = read(filename).split("\n")
+    # Skip the title (first two lines)
+    contents = contents[2:]
+
+    # Find the latest release
+    start_line, end_line = None, None
+    for i, line in enumerate(contents):
+        # if we've found the start & end, we're done
+        if all([start_line, end_line]):
+            break
+
+        # 4 dashes are a horizontal line in rST, so look for headers with 5+
+        if line.startswith("-----"):
+            if start_line is None:
+                start_line = i - 1
+                continue
+            if end_line is None:
+                end_line = i - 1
+                continue
+
+    recent_log_lines = contents[start_line:end_line]
+    recent_log = "\n".join(recent_log_lines)
+    return recent_log
+
+
+def get_main_readme_intro():
+    """Grab the intro from the README from the root of the project."""
+    readme_path = os.path.join(ROOT_DIR, "README.rst")
+    contents = read(readme_path).split(".. start-long-desc")[1:]
+    return "\n".join(contents)
+
+
+def get_long_description(package_dir):
+    """Generate the long description from the README and changelog."""
+    cl_base_path = f"reference/{package_dir}/changelog"
+    cl_file_path = os.path.join(ROOT_DIR, f"docs/src/{cl_base_path}.rst")
+    cl_url = f"{PROJECT_URLS['Documentation']}/en/latest/{cl_base_path}.html"
+
+    # When tox builds the library, it loses where the ROOT_DIR is, so we'll
+    # just skip reading files from outside the package directory while in
+    # a tox env
```

Comment:
Note that the parsing of the long description does get checked in the CI workflows ([ex](https://github.com/spotify/klio/blob/master/.github/workflows/audio.yml#L71-L72)), so the build will fail if something is b0rked. It's just not `tox` that will be able to surface it.

Note that long descriptions should be manually checked too (e.g. by running [`restview --long-description`](https://pypi.org/project/restview/) locally, and on the pypi project page after uploading).